app/models/piece.rb

```diff
@@ -12,31 +12,33 @@ def move!(x,y)
     if valid_move?(x,y)
       self.x_position = x
       self.y_position = y
+      capture!(x,y)
       return true
     else
       return false
     end
   end
 
-  #Inital Capture Logic:
-  #Need game model to test id of game
-
-  #Set initial captured status to false
-  def default_captured_status
-    self.captured = false if self.captured = nil
+  # First check if coordinates is taken
+  # return false if coordinates belongs to friendly piece
+  # return true if target piece's attributes changes
+  def capture!(x,y)
+    if position_taken?(x,y)
+      return false if friendly_piece?(x,y)
+      return true if target_piece(x,y).update!(x_position: nil, y_position: nil, captured: true)
+    end
   end
```

Comment:
I don't see anywhere in the code where you are performing the capture.  `return true if target_piece(x,y).update!(x_position: nil, y_position: nil, captured: true)` returns true if the capture has happened already, but does not do the capture.